code/weapon/weapons.cpp

```diff
@@ -6792,28 +6798,32 @@ void weapon_hit_do_sound(object *hit_obj, weapon_info *wip, vec3d *hitpos, bool
 		// play a shield hit if shields are above 10% max in this quadrant
 		if ( shield_str > 0.1f ) {
 			// Play a shield impact sound effect
-			if ( hit_obj == Player_obj ) {
+			if ( !(Use_weapon_class_sounds_for_hits_to_player) && (hit_obj == Player_obj)) {
```

Comment:
This condition should be changed to check that the shield hit sound is valid.  If it isn't, then the behavior should be as if the game_settings flag wasn't active.
```suggestion
			if ( !(Use_weapon_class_sounds_for_hits_to_player && wip->shield_impact_snd.isValid()) && (hit_obj == Player_obj)) {
```

Reply:
Never mind this suggestion